retry/regular.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package retry
+
+import (
+	"time"
+
+	"github.com/juju/utils/clock"
+)
+
+// Regular represents a strategy that repeats at regular intervals.
+type Regular struct {
+	Total time.Duration // total duration of attempt.
+	Delay time.Duration // interval between each try in the burst.
+	Min   int           // minimum number of retries; overrides Total
```

Comment:
> the Regular API is exactly the old AttemptStrategy API 

But this is a brand new package, so we should take the opportunity to make things less confusing.

> You can already do that with LimitCount. For example:

But that's a completely different mechanism than the minimum, which is why it's confusing.  It's not consistent.... so, for example, when someone looks at the fields of AttemptStrategy, they see minimum, but no maximum and think there's no way to do a maximum.  And if there's retry.LimitCount, why not retry.MinCount?